ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -184,6 +184,26 @@ boolean shouldLookupMetadataCache() {
         && options.getBlobOptions.isRawMode() == false && blobId.getBlobDataType() == BlobId.BlobDataType.METADATA);
   }
 
+  /**
+   * This helper decides if metadata must be deleted from cache or not.
+   * This mainly reduces the impact on metadata cache due to a large number of GET requests for
+   * simple blobs that do not exist.
+   * @param abortCause Reason to abort GetBlobOperation
+   * @return True if metadata must be deleted, false otherwise.
+   */
+  boolean shouldDeleteMetadata(Exception abortCause) {
+    if (abortCause != null && abortCause instanceof RouterException) {
+      switch (((RouterException) abortCause).getErrorCode()) {
```

Comment:
In this case, you could just use the previous way for switch and add break under the case right? 
Can you also add the default for the switch statement. 
And QQ: since BlobExpired is the only case we want to take action, why don't you just use a if statement.

Reply:
Sorry not clear what you mean by the first suggestion of adding a break.
And we do want to handle all three err codes and not just BlobExpired. Here is why:
1. node 1, putBlob - success
2. node 1, getBlob(range request) - success, caches metadata
3. node 2, deleteBlob 
4. node 1, getBlob(range request) - gets a blobDoesNotExist and we must clear its cache

Reply:
Also the cache is only for composite blobs. So I jsut want to filter out the simple blobs. Hence this PR.

Reply:
If you are wondering if the old code and new code are same, they're not. There is one difference. In the old, we'd attempt deleting metadata even for non-existent simple blobs. But in the new one, we'd only attempt deleting metadata for composite blobs.

Reply:
thx for the explanation, I missed that part.